### PR TITLE
Linter: don't allow version_added === version_removed

### LINKED ISSUE #3531: JSON linter: don't allow version_added === version_removed
We have a couple of places here and there where version_added is invalidly the same as version_removed; we need to update the linter to catch this.

### PR DESCRIPTION
**Pull request by vinyldarkscratch**
Fixes #3531 and all violating data.

### CODE REVIEW

**Comment by vinyldarkscratch** [comment]
Rebase complete!  Review @Elchi3?

**Comment by Elchi3** [comment]
@a2sheppy can you comment on the WebRTC changes here?

**Comment by Elchi3** [comment]
So, the added/removed nonsense comes from confluence and was submitted in this PR https://github.com/mdn/browser-compat-data/pull/3287. I think we should change it back to `null` or maybe to `false`

**Comment by vinyldarkscratch** [comment]
Sounds good to me!  I'm leaning towards setting them to `null` as it seems that the Confluence data for RTCPeerConnection isn't the most trustworthy (as stated in #3287).  Changes coming right up!

**Comment by vinyldarkscratch** [comment]
💜

File: css/properties/custom-property.json [hunk 1]
```diff
@@ -197,26 +197,12 @@
...
-                  "alternative_name": "constant"
-                }
-              ],
-              "safari_ios": [
-                {
-                  "version_added": "11.1"
-                },
-                {
-                  "version_added": "11",
-                  "version_removed": "11",
```
Comment (ExE-Boss) [inline css/properties/custom-property.json]:
```suggestion
                  "version_removed": "11.1",
```

File: css/properties/custom-property.json [hunk 2]
```diff
@@ -197,26 +197,12 @@
               "opera": {
                 "version_added": null
               },
-              "safari": [
-                {
-                  "version_added": "11.1"
-                },
-                {
-                  "version_added": "11",
-                  "version_removed": "11",
```
Comment (ExE-Boss) [inline css/properties/custom-property.json]:
```suggestion
                  "version_removed": "11.1",
```

File: api/RTCPeerConnection.json [hunk 3]
```diff
@@ -305,8 +305,7 @@
               }
             ],
             "safari": {
-              "version_added": "11",
-              "version_removed": "11"
+              "version_added": null
```
Comment (ExE-Boss) [inline api/RTCPeerConnection.json]:
```suggestion
              "version_added": "11",
              "version_removed": "11.1"
```

File: api/RTCPeerConnection.json [hunk 4]
```diff
@@ -1521,8 +1520,7 @@
...
```
Comment (ExE-Boss) [inline api/RTCPeerConnection.json]:
```suggestion
              "version_added": "11",
              "version_removed": "11.1"
```

File: api/RTCPeerConnection.json [hunk 5]
```diff
@@ -1638,8 +1636,7 @@
...
```
Comment (ExE-Boss) [inline api/RTCPeerConnection.json]:
```suggestion
              "version_added": "11",
              "version_removed": "11.1"
```

File: api/RTCPeerConnection.json [hunk 6]
```diff
@@ -1890,8 +1887,7 @@
...
```
Comment (ExE-Boss) [inline api/RTCPeerConnection.json]:
```suggestion
              "version_added": "11",
              "version_removed": "11.1"
```

File: api/RTCPeerConnection.json [hunk 7]
```diff
@@ -3728,8 +3724,7 @@
...
```
Comment (ExE-Boss) [inline api/RTCPeerConnection.json]:
```suggestion
              "version_added": "11",
              "version_removed": "11.1"
```

**Comment by vinyldarkscratch** [comment]
Ah, good catch @ExE-Boss!  I just double-checked with the Confluence data myself (I'm bad at double-checking), and can confirm that the `RTCPeerConnection` properties were indeed added in `11`, removed in `11.1`.  (I'm having a hard time tracking down the data for `custom-property`, though.)

I'll open up a new PR to insert the suggested changes.

**Comment by Elchi3** [comment]
I think it would be good to consult a different source as confluence can be wrong also. (I'm always skeptical when confluence claims something was added and removed from one version to another)

**Comment by ExE-Boss** [comment]
The `constant(…)` function&nbsp;alias was documented on&nbsp;the&nbsp;official Apple&nbsp;docs (see&nbsp;also&nbsp;#2172)

**Comment by vinyldarkscratch** [comment]
The [WebKit blog](https://webkit.org/blog/7929/designing-websites-for-iphone-x/), [Safari TP Release Notes](https://developer.apple.com/safari/technology-preview/release-notes/) (see releases 41 and 42), and [CanIUse](https://caniuse.com/#feat=css-env-function) also confirm the existence of `constant()`, which was then entirely replaced by `env()`.  The release dates of Safari TP 41 and 42 put it around `11.1`.

**Pull request merged**